### DISCUSSION TITLE
Compiling DNN leads to pybind11 error

### DISCUSSION THREAD

**Discussion by petrovicluka** [pybind/pybind11#3737] [Q&A]
First of all, when I wanted to compile my DNN with cmake../ && make -j I got this type of issue(pybind11.h missing or so):

**Cmake../:**
<https://imgur.com/38Bji0e>
**make -j:**
<https://imgur.com/D3ObL82>

After addressed issued, I have cloned the repo and successfully compiled pybind 11(as you can notice that it is found with cmake command).

However, there error remained the same.
In order to have a better understanding of a problem, I am also providing you with the CMakeLists.txt of my class-tracker which is actually trying to find pybind11:
<https://imgur.com/T7mlePv>

Thank you for you help in adnvace.

~~p.s. if you have torubles opening the links, just copy and paste them in the new tab~~

Best regards,
Luka

**Comment by henryiii** [comment]
Please paste text (inside triple tick marks), not images, and the syntax in markdown for showing a link is to either directly drop the link in like normal text, or to use `<url>`, like `<https://google.com>`. If you want a link that doesn't not show itself, then it's `[link text](link url)`.

Anyway, you are messing with and adding too much to the CMakeLists. Just use the targets and let CMake figure the rest out. Remove all lines with `PYTHON_` variables. If you need to link to something, use our CMake targets, but I think you should get it automatically everywhere anyway. You are somehow overriding the discovered Python 3.6 with an old copy (probably 2.7).
